scripts/python/phdf_diff.py

```diff
@@ -64,8 +66,120 @@ def addPath():
+            12 : Values of attribute names in Info of hdf files differ
+            13 : Meta variables (Locations, VolumeLocations, LogicalLocations, Levels) differ
+    """
+    ERROR_TIME_DIFF=10
+    ERROR_INFO_ATTRS_DIFF=12
+    ERROR_INFO_VALUES_DIFF=13
+    ERROR_META_VARS_DIFF=14
+
+    #Check in the metadata in Info is identical, except use tol for comparing "Time"
+    f0_Info = { key:value for key,value in f0.Info.items() if key != "Time" and key != "BlocksPerPE" }
+    f1_Info = { key:value for key,value in f1.Info.items() if key != "Time" and key != "BlocksPerPE" }
+
+    errTime = np.abs(f0.Time- f1.Time)
+    if errTime > tol:
+        print(f"""
+        Time of outputs differ by {f0.Time - f1.Time}
+
+        Quitting...
+        """)
+        return(ERROR_TIME_DIFF)
+    if f0_Info.keys() != f1_Info.keys():
+        print("""
+        Attributes in '/Info' of outputs differ
+
+        Quitting...
+        """)
+        return(ERROR_INFO_ATTRS_DIFF)
+
+    info_diffs = list(np.any(f0_Info[key] != f1_Info[key]) for key in f0_Info.keys())
+    infos_differ = np.any( info_diffs )
+    if infos_differ:
+        print("\n'/Info' of outputs differ\n")
+        print("Differing attributes: ",list( (key) for key in f0_Info.keys() if np.any(f0_Info[key] != f1_Info[key]))  )
+        print("\nQuitting...\n")
+        return(ERROR_INFO_VALUES_DIFF)
+    else:
+        print('  %20s: no diffs'%"Info")
+
+    # Now go through all variables in first file
+    # and hunt for them in second file.
+    #
+    # Note that indices don't match when blocks
+    # are different
+    no_meta_variables_diff = True
+
+    otherBlockIdx = list(f0.findBlockIdxInOther(f1,i) for i in range(f0.NumBlocks))
+
+    for var in set(f0.Variables+f1.Variables):
+        if var in ['Locations', 'VolumeLocations']:
+            for key in f0.fid[var].keys():
+                #Compare raw data of these variables
+                val0 = f0.fid[var][key]
+                val1 = f1.fid[var][key]
+
+                #Sort val1 by otherBlockIdx
+                val1 = val1[otherBlockIdx]
+
+                # Compute norm error, check against tolerance
+                errVal = np.abs(val0 - val1)
+                errMag = np.linalg.norm(errVal)
+                if(errMag > tol):
+                    no_meta_variables_diff = False
+                    if not quiet: print("")
+                    print(f'Metavariable {var}/{key} differs between {f0.file} and {f1.file}')
+                    if not quiet: print("")
+                else:
+                    print('  %18s/%s: no diffs'%(var,key))
+        if var in ['LogicalLocations', 'Levels']:
+            #Compare raw data of these variables
+            val0 = f0.fid[var]
+            val1 = f1.fid[var]
+
+            #Sort val1 by otherBlockIdx
+            val1 = val1[otherBlockIdx]
+
+            #As integers, they should be identical
+            if val0 != val0:
```

Comment:
```suggestion
            if val0 != val1:
```
Also, does this like work without an `.any()` ?

Reply:
It should be `np.any( val0 != val1)` for full correctness. It might be checking equivalence by reference here